File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String ret = "";
        char[] c_code = code.toCharArray();
        int mode = 0;
        
        for (int i = 0; i < c_code.length; i++) {
            if (mode == 0) {
                if (c_code[i] != '1') {
                    if (i % 2 == 0) {
                        ret += String.valueOf(c_code[i]);
                    }
                }
                if (c_code[i] == '1') {
                    mode = 1;
                }
            } else {
                if (c_code[i] != '1') {
                    if (i % 2 != 0) {
                        ret += String.valueOf(c_code[i]);
                    }
                }
                if (c_code[i] == '1') {
                    mode = 0;
                }
            }
        }
        
        if (ret.equals("")) {
            ret = "EMPTY";
        }
        
        return ret;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    mode = False
    
    for idx, i in enumerate(code):
        if i == "1":
            mode = not mode
            continue
        
        if not mode and idx % 2 == 0:
            answer += code[idx]
        elif mode and idx % 2 != 0:
            answer += code[idx]
            
    if answer == '':
        return "EMPTY"
    
    return answer